hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/multipart/MultipartDownloaderTest.java

```diff
@@ -0,0 +1,130 @@
+package org.apache.hadoop.fs.s3a.multipart;
+
+import com.google.common.base.Charsets;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.Assert.assertTrue;
+
+public class MultipartDownloaderTest {
```

Comment:
extends `AbstractS3ATestBase` and if a functional test against S3, prefixed with "ITest", not "Test", so only run in the verify phase of the build

Reply:
This one doesn't actually test S3A, just the multipart downloader.